subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeProjectComponentArtifactMetadata.java

```diff
@@ -75,6 +75,11 @@ public File getFile() {
         return artifactFile;
     }
 
+    @Override
+    public boolean isComposite() {
```

Comment:
Shouldn't be necessary. You can use `ProjectComponentIdentifier.buildIdentifier.isCurrentBuild()` to find out when something comes from a different build.